pallets/axelar-router/src/lib.rs

```diff
@@ -0,0 +1,402 @@
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The origin that is allowed to set the gateway address we accept
+		/// messages from
+		type AdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The target of the messages comming from other chains
+		type Gateway: MessageReceiver<Origin = DomainAddress>;
+
+		/// The target of the messages comming from this chain
+		type Transactor: EthereumTransactor;
+
+		/// Checker to ensure an evm account code is registered
+		type EvmAccountCodeChecker: PreConditions<(H160, H256), Result = bool>;
+	}
+
+	#[pallet::storage]
+	pub type GatewayContract<T: Config> = StorageValue<_, H160, ValueQuery>;
+
+	#[pallet::storage]
+	pub type SourceChain<T: Config> = StorageMap<_, Twox64Concat, H256, EVMChainId>;
+
+	#[pallet::storage]
+	pub type OutboundConfig<T: Config> = StorageMap<_, Twox64Concat, EVMChainId, AxelarConfig>;
+
+	//#[pallet::storage]
+	//pub type Configuration<T: Config> = StorageMap<_, Twox64Concat, H256,
+	// EVMChainId>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		InboundGatewaySet {
+			address: H160,
+		},
+		InboundChainIdSet {
+			chain_id_hash: H256,
+			chain_id: EVMChainId,
+		},
+		OutboundConfigSet {
+			config: AxelarConfig,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The given domain is not yet allowlisted, as we have no converter yet
+		NoConverterForSource,
+
+		/// A given domain expects a given structure for account bytes and it
+		/// was not given here.
+		AccountBytesMismatchForDomain,
+
+		/// Router not found.
+		RouterNotFound,
+
+		/// Emit when the evm account code is not registered
+		ContractCodeNotMatch,
+	}
+
+	impl<T: Config> Pallet<T> {
+		fn weight_for_set_method() -> Weight {
+			Weight::from_parts(10_000_000, 128).saturating_add(RocksDbWeight::get().writes(1))
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(Pallet::<T>::weight_for_set_method())]
+		#[pallet::call_index(0)]
+		pub fn set_inbound_gateway(origin: OriginFor<T>, address: H160) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			GatewayContract::<T>::set(address);
+
+			Self::deposit_event(Event::<T>::InboundGatewaySet { address });
+
+			Ok(())
+		}
+
+		#[pallet::weight(Pallet::<T>::weight_for_set_method())]
+		#[pallet::call_index(1)]
+		pub fn set_inbound_chain(
+			origin: OriginFor<T>,
+			chain_id_hash: H256,
+			chain_id: EVMChainId,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			SourceChain::<T>::insert(chain_id_hash, chain_id);
+
+			Self::deposit_event(Event::<T>::InboundChainIdSet {
+				chain_id_hash,
+				chain_id,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::weight(Pallet::<T>::weight_for_set_method())]
+		#[pallet::call_index(2)]
+		pub fn set_outbound_config(
```

Comment:
The router is configured here by `chain_id` instead of domain because a router can never be a Centrifuge domain. The gateway will no longer configure routers, will only handle the routers id.